internal/experiments/experiments_remote_execute_api.go

```diff
@@ -0,0 +1,205 @@
+// a few domains ("https://google.com", "https://linkedin.com", "https://openai.com/") and responds with a success based on the success of those calls.
+// The source can be found at internal/executor/container_examples/egress_server
+type RemoteExecuteAPI struct {
+	Image                string `yaml:"image"`
+	ImageParameters      string `yaml:"image_parameters"`
+	TargetPort           int32  `yaml:"target_port"`
+	Path                 string `yaml:"path"`
+	ServiceAccountName   string `yaml:"service_account_name"`
+	LocalPort            int32  `yaml:"local_port"`
+	ExpectedHTTPResponse int    `yaml:"expected_response"`
+	ExpectedBody         string `yaml:"expected_body"`
+}
+
+func (p *RemoteExecuteAPIExperimentConfig) Type() string {
+	return "remote_execute_api"
+}
+
+func (p *RemoteExecuteAPIExperimentConfig) Description() string {
+	return "This experiment runs a deployment based on a configurable image and then verifies based off of API calls to that image"
+}
+func (p *RemoteExecuteAPIExperimentConfig) Technique() string {
+	return categories.MITRE.Execution.NewContainer.Technique
+}
+func (p *RemoteExecuteAPIExperimentConfig) Tactic() string {
+	return categories.MITRE.Execution.NewContainer.Tactic
+}
+func (p *RemoteExecuteAPIExperimentConfig) Framework() string {
+	return string(categories.Mitre)
+}
+
+func (p *RemoteExecuteAPIExperimentConfig) Run(ctx context.Context, client *k8s.Client, experimentConfig *ExperimentConfig) error {
+	var config RemoteExecuteAPIExperimentConfig
+	yamlObj, _ := yaml.Marshal(experimentConfig)
+	err := yaml.Unmarshal(yamlObj, &config)
+	if err != nil {
+		return err
+	}
+
+	executorConfig := executor.NewExecutorConfig(
+		config.Metadata.Name,
+		config.Metadata.Namespace,
+		config.Parameters.Image,
+		config.Parameters.ImageParameters,
+		config.Parameters.ServiceAccountName,
+		config.Parameters.TargetPort,
+		config.Parameters.LocalPort,
+	)
+
+	err = executorConfig.Deploy(ctx, client.Clientset)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *RemoteExecuteAPIExperimentConfig) Verify(ctx context.Context, client *k8s.Client, experimentConfig *ExperimentConfig) (*verifier.Outcome, error) {
+	var config RemoteExecuteAPIExperimentConfig
+	yamlObj, _ := yaml.Marshal(experimentConfig)
+	err := yaml.Unmarshal(yamlObj, &config)
+	if err != nil {
+		return nil, err
+	}
+
+	executorConfig := executor.NewExecutorConfig(
+		config.Metadata.Name,
+		config.Metadata.Namespace,
+		config.Parameters.Image,
+		config.Parameters.ImageParameters,
+		config.Parameters.ServiceAccountName,
+		config.Parameters.TargetPort,
+		config.Parameters.LocalPort,
+	)
+
+	stopChan := make(chan struct{}, 1)
+	readyChan := make(chan struct{}, 1)
+	errch := make(chan error, 1)
+	out, errOut := new(bytes.Buffer), new(bytes.Buffer)
+
+	forwardConfig := executor.PortForwardConfig{
+		StopCh:  stopChan,
+		ReadyCh: readyChan,
+		Out:     out,
+		ErrOut:  errOut,
+	}
```

Comment:
These would then be in the `executorConfig` and you no longer need to pass in `forwardConfig` to `OpenLocalPort`